gpbft/chain.go

```diff
@@ -48,6 +54,27 @@ type TipSet struct {
 	Commitments [32]byte
 }
 
+var EMPTY_COMMITMENTS = [32]byte{}
+
+func (ts *TipSet) Validate() error {
+	if len(ts.Key) == 0 {
+		return errors.New("tipset key must not be empty")
+	}
+	if len(ts.Key) > TIPSET_KEY_MAX_LEN {
```

Comment:
I'd consider limiting this as well in cbor-gen as well with a ``﻿`cborgen:"maxlen=38"`` struct tag.

Reply:
Or _only_ limiting it in cbor-gen.

Reply:
Added a struct tag too.